app/slashcommands/command_invite.go

```diff
@@ -38,58 +38,150 @@ func (*InviteProvider) GetCommand(a *app.App, T i18n.TranslateFunc) *model.Comma
+
+	targetUsers := make([]*model.User, 0, 1)
+	targetChannels := make([]*model.Channel, 0)
+
+	if len(splitMessage) == 1 {
+		userProfile := i.getUserProfile(a, strings.TrimPrefix(splitMessage[0], "@"))
+		if userProfile == nil {
+			return targetUsers, targetChannels, &model.CommandResponse{
+				Text:         args.T("api.command_invite.missing_user.app_error"),
 				ResponseType: model.CommandResponseTypeEphemeral,
 			}
 		}
+		targetUsers = append(targetUsers, userProfile)
 	} else {
-		channelToJoin, err = a.GetChannel(c, args.ChannelId)
+		for _, msg := range splitMessage {
+			if len(msg) == 0 {
+				continue
+			}
+
+			switch msg[0] {
+			case '@':
+				targetUsername := strings.TrimPrefix(msg, "@")
+				userProfile := i.getUserProfile(a, targetUsername)
+				if userProfile == nil {
+					return targetUsers, targetChannels, &model.CommandResponse{
+						Text:         args.T("api.command_invite.missing_user.app_error"),
+						ResponseType: model.CommandResponseTypeEphemeral,
+					}
+				}
+				targetUsers = append(targetUsers, userProfile)
+			case '~':
+				targetChannelName := strings.TrimPrefix(msg, "~")
+				mlog.Debug("targetChannelName", mlog.String("targetChannelName", targetChannelName))
+				channelToJoin, err := a.GetChannelByName(c, targetChannelName, args.TeamId, false)
+				if err != nil {
+					return targetUsers, targetChannels, &model.CommandResponse{
+						Text: args.T("api.command_invite.channel.error", map[string]any{
+							"Channel": targetChannelName,
+						}),
+						ResponseType: model.CommandResponseTypeEphemeral,
+					}
+				}
+				targetChannels = append(targetChannels, channelToJoin)
+			default:
+				channelToJoin, err := a.GetChannelByName(c, msg, args.TeamId, false)
+				if err != nil {
+					return targetUsers, targetChannels, &model.CommandResponse{
+						Text: args.T("api.command_invite.channel.error", map[string]any{
+							"Channel": msg,
+						}),
+						ResponseType: model.CommandResponseTypeEphemeral,
+					}
+				}
+				targetChannels = append(targetChannels, channelToJoin)
+			}
+		}
+	}
+
+	if len(targetUsers) == 0 {
+		return targetUsers, targetChannels, &model.CommandResponse{
+			Text:         args.T("api.command_invite.missing_user.app_error"),
+			ResponseType: model.CommandResponseTypeEphemeral,
+		}
+	}
+
+	if len(targetChannels) == 0 {
+		channelToJoin, err := a.GetChannel(c, args.ChannelId)
 		if err != nil {
-			return &model.CommandResponse{
+			return targetUsers, targetChannels, &model.CommandResponse{
 				Text:         args.T("api.command_invite.channel.app_error"),
 				ResponseType: model.CommandResponseTypeEphemeral,
 			}
 		}
+		targetChannels = append(targetChannels, channelToJoin)
 	}
 
+	return targetUsers, targetChannels, nil
+}
+
+func (i *InviteProvider) getUserProfile(a *app.App, username string) *model.User {
+	userProfile, nErr := a.Srv().Store().User().GetByUsername(username)
+	if nErr != nil {
+		mlog.Error(nErr.Error())
```

Comment:
Gentle reminder to check our logging [principles](https://developers.mattermost.com/contribute/server/style-guide/#log-levels) before merging this change.